presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotPushdownUtils.java

```diff
@@ -163,6 +185,56 @@ public ExpressionType getExpressionType()
         return nodeBuilder.build();
     }
 
+    /**
+     * Try to push down query like: `SELECT count(distinct $COLUMN) FROM myTable` to Pinot as `SELECT distinctCount($COLUMN) FROM myTable`.
+     * This function only handles the case of an AggregationNode (COUNT on $COLUMN) on top of an AggregationNode(of non-aggregate on $COLUMN).
+     *
+     * @param nodeBuilder
+     * @param aggregationNode
+     * @param outputColumn
+     * @param aggregation
+     * @return true if push down successfully otherwise false.
+     */
+    private static boolean handlePushDownSingleDistinctCount(ImmutableList.Builder<AggregationColumnNode> nodeBuilder, AggregationNode aggregationNode, VariableReferenceExpression outputColumn, AggregationNode.Aggregation aggregation)
+    {
+        if (aggregation.getCall().getDisplayName().equalsIgnoreCase(COUNT_FUNCTION_NAME)) {
+            List<RowExpression> arguments = aggregation.getCall().getArguments();
+            if (arguments.size() == 1) {
+                RowExpression aggregationArgument = arguments.get(0);
+                // Handle the case of Count Aggregation on top of a Non-Agg GroupBy Aggregation.
+                if (aggregationNode.getSource() instanceof AggregationNode) {
+                    AggregationNode sourceAggregationNode = (AggregationNode) aggregationNode.getSource();
+                    Set<String> sourceAggregationGroupSet = getGroupKeys(sourceAggregationNode.getGroupingKeys());
+                    Set<String> aggregationGroupSet = getGroupKeys(aggregationNode.getGroupingKeys());
+                    aggregationGroupSet.add(aggregationArgument.toString());
+                    if (sourceAggregationGroupSet.containsAll(aggregationGroupSet) && aggregationGroupSet.containsAll(sourceAggregationGroupSet)) {
+                        //(sourceAggregationNode.getGroupingKeys().containsAll(aggregationNode.getGroupingKeys()))
+                        //aggregationArgument.equalsIgnoreCase(sourceAggregationNode.getGroupingKeys().get(0).getName()))
+                        nodeBuilder.add(
+                                new AggregationFunctionColumnNode(
+                                        outputColumn,
+                                        new CallExpression(
+                                                DISTINCT_COUNT_FUNCTION_NAME,
+                                                aggregation.getFunctionHandle(),
+                                                BigintType.BIGINT,
```

Comment:
Why is the return type unconditionally a BIGINT ?

Reply:
cause Pinot always returns a LONG value for this

Reply:
Hmm. I wonder if the return type should match the return type expected in the presto plan ? Perhaps we should only push down if the return type is a long ? 

I can't think of count-distinct returning anything other than a Long in presto, but just in case.

Reply:
will add the check.

Reply:
I will passing the original type from the aggregation call and let PlanType to translate the return value.